scala-mode-syntax.el

```diff
@@ -283,11 +283,12 @@
   (concat "\\(^\\|[^`'_]\\)\\(" scala-syntax:value-keywords-unsafe-re "\\)"))
 
 (defconst scala-syntax:other-keywords-unsafe-re
-  (regexp-opt '("abstract" "case" "catch" "class" "def" "do" "else" "extends"
-                "final" "finally" "for" "forSome" "if" "implicit" "import"
-                "lazy" "match" "new" "object" "override" "package" "private"
-                "protected" "return" "sealed" "throw" "trait" "try" "type"
-                "val" "var" "while" "with" "yield" "inline") 'words))
+  (regexp-opt '("abstract" "case" "catch" "class" "def" "do" "else" "enum"
+                "export" "extends" "final" "finally" "for" "given" "forSome"
+                "if" "implicit" "import" "lazy" "match" "new" "object"
+                "override" "package" "private" "protected" "return" "sealed"
+                "then" "throw" "trait" "try" "type" "val" "var" "while"
+                "with" "yield" "inline") 'words))
```

Comment:
There are also new soft keyword (`inline` is one of them) https://dotty.epfl.ch/docs/internals/syntax.html#soft-keywords

Reply:
@enzief good point. `inline` was pre-existing in this list, though; may have to look more fully into how that changed.

@hvesalai is there an existing strategy for "soft" keywords?

Reply:
The what the what and the what? What is a "soft" keyword?

Reply:
I haven't looked at it yet but suspect that it is a keyword which is only reserved in certain subsections of the grammar. `set` and `get` behave this way in C#, if memory serves.

Reply:
Ok. No support for that in emacs. My advice is not to even try. Just make them normal keywords, i.e. they will be highlighted as such regardless of the context.

Reply:
Sounds reasonable to me. I'll get the other "soft keywords" (except the symbolic ones) added to this list when I get a chance.